lib/user.js

```diff
@@ -1,87 +1,92 @@
-        mediator.publish('done:' + topicLoad + ':' + id, user);
-      } else {
-        mediator.publish('error:' + topicLoad + ':' + id, 'No such user');
+    store.find(id, function(err, user) {
+      if (err) {
+        return mediator.publish('error:' + topicLoad + ':' + id, 'No such user');
       }
-    }, 0);
+      mediator.publish('done:' + topicLoad + ':' + id, user);
+    });
   });
 
   var topicUsernameLoad = 'wfm:user:username:read';
   console.log('Subscribing to mediator topic:', topicUsernameLoad);
   mediator.subscribe(topicUsernameLoad, function(username) {
-    setTimeout(function() {
-      var user = _.find(users, function(_user) {
-        return _user.username === username;
-      });
-      if (user) {
-        mediator.publish('done:' + topicUsernameLoad + ':' + username, user);
-      } else {
-        mediator.publish('error:' + topicUsernameLoad + ':' + username, 'No such user');
+    store.byUsername(username, function(err, user) {
+      if (err) {
+        return mediator.publish('error:' + topicUsernameLoad + ':' + username, 'No such user');
       }
-    }, 0);
+      mediator.publish('done:' + topicUsernameLoad + ':' + username, user);
+    });
   });
 
   var topicSave = 'wfm:user:update';
   console.log('Subscribing to mediator topic:', topicSave);
   mediator.subscribe(topicSave, function(user) {
-    setTimeout(function() {
-      var index = _.findIndex(users, function(_user) {
-        return _user.id === user.id;
-      });
-      users[index] = user;
-      console.log('Saved user:', user);
-      mediator.publish('done:' + topicSave + ':' + user.id, user);
-    }, 0);
+    const id = user.id;
+    store.update(user.id, user, function(err, user) {
+      if (err) {
+        return mediator.publish('error:' + topicSave + ':' + id, 'No such user');
+      }
+      mediator.publish('done:' + topicSave + ':' + id, user);
+    });
   });
 
   var topicCreate = 'wfm:user:create';
   console.log('Subscribing to mediator topic:', topicCreate);
   mediator.subscribe(topicCreate, function(user) {
-    setTimeout(function() {
-      user.id = shortid.generate();
-      users.push(user);
-      console.log('Created user:', user);
-      mediator.publish('done:' + topicCreate + ':' + user.createdTs, user);
-    }, 0);
+    store.create(user, function(err, user) {
+      const id = user.id;
+      if (err) {
+        return mediator.publish('error:' + topicCreate + ':' + id, 'No such user');
+      }
+      mediator.publish('done:' + topicCreate + ':' + id, user);
+    });
+  });
+
+  var topicVerify = 'wfm:user:verifyPassword';
+  console.log('Subscribing to mediator topic:', topicVerify);
+  mediator.subscribe(topicVerify, function(id, pwd) {
+    store.verifyPassword(id, pwd, function(err, passwordCorrect) {
+      if (err) {
+        return mediator.publish('error:' + topicVerify + ':' + id, 'User not found');
+      }
+      mediator.publish('done:' + topicVerify + ':' + id, passwordCorrect);
+    });
```

Comment:
@nialldonnellyfh was discussing with @witmicko and we were thinking about having `demo-auth` provide another topic for validating the user's password in `raincatcher-user` (for RAINCATCH-398) since -auth has all the hashing logic for now. Is that okay in your view?

We'd eventually like to refactor this stuff out to `raincatcher-user` since `demo-auth` seems have a lot of functionality (data manipulation mostly) that we'd like to provide as part of the self-contained user module, just with the caveat of tying the running environment with the auth services feature of the platform.